File: app/models/benchmark/BenchmarkQuestion.scala
package models.benchmark

final case class BenchmarkQuestion(
                                    categories: Option[List[String]],
                                    choices: List[BenchmarkQuestionChoice],
                                    concept: Option[String],
                                    correctChoiceLabel: String,
                                    id: String,
                                    datasetId: String,
                                    prompts: List[BenchmarkQuestionPrompt],
                                    `type`: Option[BenchmarkQuestionType]
                                  )

File: test/integration/cypress/support/models/benchmark/BenchmarkQuestion.ts
enum BenchmarkQuestionPromptType {
  Goal = "Goal",
  Observation = "Observation",
  Question = "Question",
}

interface BenchmarkQuestionPrompt {
  text: string;
  type: BenchmarkQuestionPromptType;
}

export interface BenchmarkQuestion {
  choices: {
    id: string;
    identifier?: string;
    position: number;
    text: string;
  }[];
  concept?: string;
  correctChoiceId: string;
  datasetId: string;
  id: string;
  prompts: BenchmarkQuestionPrompt[];
}
